classes/Hook.php

```diff
@@ -1262,11 +1269,17 @@ private static function getMethodName(string $hookName): string
      */
     public static function getHookStatusByName($hook_name): bool
     {
-        $sql = new DbQuery();
-        $sql->select('active');
-        $sql->from('hook', 'h');
-        $sql->where('h.name = "' . pSQL($hook_name) . '"');
+        if (!Cache::isStored('hook_active')) {
+            $sql = new DbQuery();
+            $sql->select('name');
+            $sql->from('hook', 'h');
+            $sql->where('h.active = 1');
+            $active_hooks = Db::getInstance()->executeS($sql);
+            if (!empty($active_hooks)) {
+                Cache::store('hook_active', array_column($active_hooks, 'name'));
+            }
+        }
 
-        return (bool) Db::getInstance()->getValue($sql);
+        return in_array($hook_name, Cache::retrieve('hook_active'));
```

Comment:
```suggestion
        return in_array($hook_name, Cache::retrieve('hook_active'));
```

Please check that `Cache::retrieve('hook_active')` is an array

Reply:
Instead of testing on cache read, it is much better to test on cache write. That way, the correct type on cache read is guarateed.

Reply:
@lmeyer1 this is the reason why unit tests are failing, if you do:

```
$active_hooks = Cache::retrieve('hook_active') ?? [];
return in_array($hook_name, $active_hooks);
```

tests are green :-) In unit tests cache is not set

Reply:
@kpodemski 
Thank you. But this will always return false if cache is disabled... Shouldn't it return true if module is active, even in test context?
Now that I understand what's wrong, I hope to get it right.